Source/BusinessMomentsAdapters/Backend/domain/miners/Miner.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { AggregateRoot, aggregateRoot } from '@dolittle/sdk.aggregates';
+import { Guid } from '@dolittle/rudiments';
+import { EntityAssociatedWithMiner, ImperativeEmbedDefined } from '../../events/miners';
+
+@aggregateRoot('f4d99676-cd37-4e7e-8a03-95cd29b5927d')
+export class Miner extends AggregateRoot {
+
+    associateEntity(entityId: Guid) {
+        this.apply(new EntityAssociatedWithMiner(entityId.toString()));
```

Comment:
Should we allow duplicate associations?

Reply:
Good question. 

From a domain perspective, we also need to decide if associations can only happen once and never changed.
If we allow for changing the entity association, I see two approaches to this: 

1. Introduce a rule that will break if you associate with the same entity twice
2. Silently ignore if one associates with the same entity twice (not apply an event)

Reply:
Hmmm...

I think we should avoid producing duplicate events when associating entity with miner. I don't see any benefits of the event being duplicated. Do you? I'm just asking in case I'm ignoring some details.

I don't think it's a good solution to silently ignore duplicated associations. The rule will give a very clear hint to the developer that the entity has already been associated. This can help when debugging on the consumer side.

Even though we don't have a use case for it yet, I have a feeling that we'll need to be able to remove associations too.

I think we should go for option 1.